checkstyle-checks/src/main/resources/sonatype/checkstyle-configuration.xml

```diff
@@ -25,8 +25,14 @@
 
   <module name="LineLength">
     <property name="max" value="120"/>
-    <!-- Allow longer lines in block comments and whole string literals -->
-    <property name="ignorePattern" value='^ +\*|"[^"]{100,}"'/>
+    <!--
+        Allow longer lines in:
+        - package statements
+        - import statements
+        - block comments
+        - whole string literals
+    -->
+    <property name="ignorePattern" value='^package |^import |^ +\*|"[^"]{100,}"'/>
```

Comment:
Thinking out loud: this part `^ +\*` wouldn't match the start of a comment, like `/*` or `/**`, right?

I'm guessing that's ok as we don't usually have something on that line for a block comment
```java
/* few people put text on this part of the block comment
 * but putting it here is common
 */

/** I suppose you could do this for an inline javadoc and it could be a long line, but in that case you could just make it a longer, more typical block javadoc comment */
```

What was the use case for block comments?

Reply:
That one was put in place in my previous iteration on this a while ago, but I believe we were seeing things like tables and URLs in comments, and especially accompanying tests (maybe?). 
```
  /**
   * Packagist pattern for a valid package "name" https://github.com/composer/composer/blob/54aac000ba78f8d6848fe4602710d41c3758c5e2/src/Composer/Package/Loader/ValidatingArrayLoader.php#L57
   */
  private static final String VENDOR_AND_PACKAGE_REGEX =
      "(?<vendor>[A-Za-z0-9][A-Za-z0-9_.-]*)/(?<package>[A-Za-z0-9][A-Za-z0-9_.-]*)";
```

Reply:
these things won't be bulletproof, since it's just regex, but they'll help take the edge off some of the rules. I tried to allow long whole strings, but once you stick an escaped quote in there (`\"`), this pattern can't help you.